flagsmith/analytics.py

```diff
@@ -40,20 +51,27 @@ def flush(self):
 
         if not self.analytics_data:
             return
-        session.post(
-            self.analytics_endpoint,
-            data=json.dumps(self.analytics_data),
-            timeout=self.timeout,
-            headers={
-                "X-Environment-Key": self.environment_key,
-                "Content-Type": "application/json",
-            },
-        )
-
-        self.analytics_data.clear()
-        self._last_flushed = datetime.now()
+
+        try:
+            self.session.post(
+                self.analytics_endpoint,
+                data=json.dumps(self.analytics_data),
+                timeout=self.timeout,
+                headers={
+                    "X-Environment-Key": self.environment_key,
+                    "Content-Type": "application/json",
+                },
+            )
+            self.analytics_data.clear()
+        except requests.ConnectionError as e:
```

Comment:
Note that error will only be logged if we are using `request.session` because future session just keeps them silent